File: src/main/java/leetcode/leetcode876.java
package leetcode;

import util.ListNode;

public class leetcode876 {
	public ListNode middleNode(ListNode head) {
		ListNode slow = head;
		ListNode fast = head;
		while(fast.next!=null){
			slow = slow.next;
			fast = fast.next;
			if(fast.next!=null){
				fast = fast.next;
			}else{
				break;
			}
		}
		return slow;
    }
}

File: src/main/java/JavaScript/tree_level_visit/leet_code/leetcode-876.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var middleNode = function(head) {
  var half_node = head;
  var final_node = head;
  while(final_node.next!=null){
    if(final_node.next.next==null){
      half_node = half_node.next;
      final_node = final_node.next;
    }else{
      half_node = half_node.next;
      final_node = final_node.next.next
    }
  }
  return half_node
};

function ListNode(val) {
  this.val = val;
  this.next = null;
}
